### DISCUSSION TITLE
still use MyCroft.  This day: , 'Connection to api.mycroft.ai timed out.' - whats next?

### DISCUSSION THREAD

**Discussion by geetee24** [OpenVoiceOS/ovos-buildroot#217] [Q&A] [answered]
Hi.   Any idea why today mycroft message , 'Connection to api.mycroft.ai timed out.'?   Is my MyCroft now totally dead?  If so any ideas on if OVos can consume all my MC skills?  If so is there an upgrade guide?

**Comment by cooljimy84** [comment]
Mycroft.ai is giving a 404, which prob means they are gone. Ive tested the mk1 image (on a sdcard) and it works, though is quite slow, but really not bad as OVOS have refactored the code.

**Comment by geetee24** [comment]
hi.  what do you mean:

Ive tested the mk1 image (on a sdcard) and it works, though is quite slow, but really not bad as OVOS have refactored the code.

**Reply by cooljimy84** [reply]
I booted my Mycroft mk1 with the ovos mk1 testing img and it ran well. Not as quick as the mk1 used to, but from what I have read ovos have done loads of work to port code and get it up to date.

**Comment by geetee24** [comment]
But supposedly OVOS will not run any Microsoft  skills

**Reply by mikejgray** [reply]
Not sure what you mean - OVOS will maintain compatibility with Mycroft skills until version 0.1.0. Are there particular skills you're having issues with? Media skills are not compatible but everything else should work.

**Reply by geetee24** [reply]
Hi Mike. What happens when zero.10 comes out? When is that going to happen? And then all my skills will stop working? What do I do then? Do I have to throw out my OVOS unit or can I leave it at a version right before zero.10 that my skills can continue to work. is there a migration guy that tells me what I have to do to my skills to get them to work after 0.1.0

**Reply by JarbasAl** [reply]
when 0.1.0 comes out anything that imports from `mycroft` module will stop working, at that point you must update your skills to use the new imports from ovos packages. should be all simple changes, the projen project from Mike linked earlier will do most of it for you automatically depending on the complexity of your skills

if you stick to 0.0.8  (the next stable release, currently in testing for stability, no new changes coming) then old mycroft skills will keep working, you arent forced to update to 0.1.0 until you want too.

**Comment by geetee24** [comment]
but after 0.1.0 they will all break :(

**Reply by JarbasAl** [reply]
which skills are you missing that dont have a updated alternative?

**Comment by geetee24** [comment]
i have a bunch of custom skills i wrote.   will I have to rewrite them for
> 0.1.0?

**Reply by JarbasAl** [reply]
https://github.com/mikejgray/ovos-skill-projen should help you migrate your own skills. not a lot of changes needed!

**Comment by geetee24** [comment]
I have a SEED audio daughterboard with built-in Mike and amplifier, so am I
gonna have to install that from scratch also? Do I install OVOS over my
existing PICROFT image, which of course has all of my skills in it or do I
have to somehow blow away my existing PICROFT and then copy all my skills
back in? Is there any migration guy that talks about the SEED daughterboard
card?

**Comment by JarbasAl** [comment, marked as answer]
If you need to keep a picroft alive, you can connect it to the personal backend to trick it into thinking mycroft servers are up

https://openvoiceos.github.io/ovos-technical-manual/personal_backend/#classic-mycroft-devices


we do not have a migration guide unfortunately, a picroft image can not be upgraded in place, only replaced with a fresh install

**Reply by JarbasAl** [reply]
side note: while this is expected to work, we do not explicitly provide support for classic mycroft devices, our recommendation is to move away from classic mycroft completely

any issues in the personal backend repo will be addressed, but noone in the OVOS team has used a picroft for years at this point, it is completely untested

**Comment by geetee24** [comment]
Hi. I read the link regarding the personal backend. And that sounds
perfect. Although the document is confusing to me because it the document
seems to talk about the OVOS backend in great detail, but it seems to
apply. I want to use the PICROFT back and all I have to do is make one
change to the JSON file and Presto it’s working, is that correct?

**Reply by JarbasAl** [reply]
the document is about the OVOS backend specifically, hence it going into detail :P since 2018 the personal backend has been the only way to trick picrofts to run offline

you need to install and run the backend somewhere in your home network, or on the picroft device itself, and then point mycroft.conf to it. 

The docs linked are about how to manage the backend itself, not about mycroft

**Comment by geetee24** [comment]
And my MYCROFT on my raspberry PI of course runs in a virtual environment.
But your documentation doesn’t talk about that if I wanted to use the OVOS
manager do I install it inside the environment or outside of the
environment?

**Comment by geetee24** [comment]
I’d like to run the backend on the PICROTFT device itself, so do I install
it within the virtual environment of the PICROFT or outside of the
environment?

**Reply by JarbasAl** [reply]
doesnt matter, picroft just talks to it via http, it can be anywhere you want, in a venv, in a docker container, in another machine.....

**Comment by geetee24** [comment]
Where do I find the file?
identity2.json

**Reply by JarbasAl** [reply]
i think it was ~/.mycroft/identity  but not sure anymore, check the mycroft docs, its been years since i touched that :sweat_smile:

**Comment by geetee24** [comment]
All the micro docs are off-line

**Comment by geetee24** [comment]
i am now totally lost after readings the docs :(

https://openvoiceos.github.io/ovos-technical-manual/personal_backend/#classic-mycroft-devices

1. what command do i issue to install the back end into my windows machine.
the above link refs installing the back end manager UI but doesnt mention
how to install the back end itself.

**Comment by geetee24** [comment]
more confusion.  on https://github.com/OpenVoiceOS/ovos-personal-backend

it says

database can be sqlite or mysql eg. mysql+mysqldb://
***@***.***/test?ssl_ca=/path/to/ca.pem&ssl_cert=/path/to/client-cert.pem&ssl_key=/path/to/client-key.pem

-well ...... give us some guidance. What is already installed? How do we
choose? How do we install?

argh..........

**Comment by geetee24** [comment]
on windows in admin command prompt: this fails.

c:\>docker run -p 8086:6712 -d --restart always --name local_backend
ghcr.io/openvoiceos/local-backend:dev
docker: error during connect: In the default daemon configuration on
Windows, the docker client must be run with elevated privileges to
connect.: Post "http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.24/containers/create?name=local_backend":
open //./pipe/docker_engine: The system cannot find the file specified.
See 'docker run --help'.

**Comment by geetee24** [comment]
still totally lost following your docs :(

SO I run the docker image on my windows machine and then what?  What is the
comment "the ...yml could look like this? Is a YAML needed?  if so, what
are the exact steps?


Docker

There is also a docker container you can use

docker run -p 8086:6712 -d --restart always --name local_backend
ghcr.io/openvoiceos/local-backend:dev

a docker-compose.yml could look like this

version: '3.6'services:
  # ...
  ovosbackend:
    container_name: ovos_backend
    image: ghcr.io/openvoiceos/local-backend:dev
    # or build from local source (relative to docker-compose.yml)
    # build: ../ovos/ovos-personal-backend/.
    restart: unless-stopped
    ports:
      - "6712:6712"                                              #
default port backend API
      - "36535:36535"                                            #
default port backend-manager
    volumes: # <host>:<guest>:<SELinux flag>
      - ./ovos/backend/config:/root/.config/json_database:z      #
shared config directory
      - ./ovos/backend/data:/root/.local/share/ovos_backend:Z    #
shared data directory
      # set `data_path` to `/root/.local/share/ovos_backend`

**Comment by geetee24** [comment]
I am running docker run -p 8086:6712 -d --restart always --name
local_backend ghcr.io/openvoiceos/local-backend:dev
on my windows machine.

what are the precise steps needed to point my picroft to it?

so confusing lack of completeness in the docs

**Reply by JarbasAl** [reply]
just update your picroft mycroft.conf to use the backend, replace the url with your own....
```
{
  "server": {
    "url": "http://0.0.0.0:6712"
  }
}
```

or just use a public server, i strongly recommend against it, you are supposed to self host, this is for demo purposes https://ovospublicbackend.ziggyai.online/

**Comment by geetee24** [comment]
I now have the ovos personal backed in docker on windows running.   I see it at the proper port....

also mycroft also can ping it and curl it.

however mycroft is not able to understand my voice prompts.  like what time is it.   appears the link to ovos is not working. also mycroft speaks to me and tells me "> It seems I cannot connect to the Mycroft servers. Please wait few minutes before trying to speak to me.                          

any ideas please?

**Reply by mikejgray** [reply]
Personal backend hasn't been maintained much in a while. If you're able, I'd strongly recommend switching to OVOS. Most skills should be supported from a Mycroft with no change, and the majority from Mycroft have already been ported to OVOS with improvements.

**Comment by geetee24** [comment]
thanks but then I have to remember how to configure my respeaker.

**Comment by geetee24** [comment]
If I install OVOS from scratch on a raspberry PI does it use the devices
microphone and line out by default?

**Reply by builderjer** [reply]
If you use the [raspOVOS headless](https://ovosimages.ziggyai.online/raspbian/newest), most devices work out of the box.
Tested with
- RPi3b/4
  - ReSpeaker 4mic square - USB speaker
  - ReSpeaker 2mic
    - speaker connection works out of the box
    - 3.5mm audio out needs unmuted in `alsamixer`
      - Working on this issue
  - USB Mic
    - PS3 Eye
    - BlueSnowball
  - Google AIY Voicebonnet V2
    - Everything out of the box except the button
  - Google AIY VoiceHAT V1
    - With `dtoverlay=googlevoicehat-soundcard` added to `/boot/firmware/config.txt`
    
If you have any other issues, or a HAT that does NOT work, please let me know.

*edit*
Mycroft Mark1 is coming with the next build

**Comment by geetee24** [comment]
on rp 4, which do i want:?

raspOVOS-gui-dev.zip
<https://ovosimages.ziggyai.online/raspbian/newest/raspOVOS-gui-dev.zip>
                              13-Feb-2024 16:39
2308661561raspOVOS-headless-dev.zip
<https://ovosimages.ziggyai.online/raspbian/newest/raspOVOS-headless-dev.zip>

**Reply by builderjer** [reply]
I would suggest the headless image as it is working better.  If you do not need the GUI

The GUI image is working on everything but the Mark2 and only the mic does not work on it.  That one is high on the priority list

**Comment by geetee24** [comment]
which is the latest stable version (not dev)?

**Reply by builderjer** [reply]
I am ready to make the headless image stable, but for now, everything is bleeding edge, but works well

**Comment by geetee24** [comment]
nothing in stable..... https://ovosimages.ziggyai.online/raspbian/stable/

the confusion in the docs is widespread.

**Comment by geetee24** [comment]
what is the latest STABLE image?

**Reply by builderjer** [reply]
I am in the process of organizing the images into a "stable" "testing" and "dev" branch, but is just now getting started

**Reply by builderjer** [reply]
The image that is building now will go into the "testing" directory, which is the step before stable.

**Comment by geetee24** [comment]
is there docs on how to update the OVOS code once you get into stable?

**Reply by builderjer** [reply]
I don't currently support a direct upgrade, it would be a new image.

That is on the todo list, but Neon is just getting it right and Buildroot images are working on that also.

I mainly built this image as a development image, but there are a lot of users that run it now, and it has WAY evolved from it conception less than a year ago.

**Comment by geetee24** [comment]
what are the passwords for the image to SSH into it?

**Reply by builderjer** [reply]
ssh ovos@raspovos
pw ovos

**Comment by geetee24** [comment]
how do i set the SID info?

**Comment by geetee24** [comment]
it speaks failed to verify creds. please try again.    where are your setup
url that tells me how to specify the SID ands SSH enable.

**Reply by builderjer** [reply]
Did you use setup your wifi on another device?
Are you using WiFi6?

*edit* 
connect to SSID "OVOS"
open page openvoiceos.com
enter your credentials

**Reply by builderjer** [reply]
ssh is enabled by default

**Comment by geetee24** [comment]
huh?

how do i tell the image to use my sid at 5ghz?

**Reply by builderjer** [reply]
When you first boot your image, it will take some time.  It is resizing the filesystem and doing some setup.  It then reboots.

Shortly after that, you will hear a prompt that "A temporary hotspot has been created called OVOS"
Connect your phone to that SSID (NOT the SSID of your home network)
Browse to https://openvoiceos.com
Enter your credentials
In a couple of minutes, your device is ready.

**Reply by builderjer** [reply]
As long as your home network is NOT using WiFi6.  The plugin has issues with that.

**Comment by geetee24** [comment]
i give up.  you cant tell me how to assign my SID and wifi PW to your image
so the image when boots up cant connect to wifi and the volume on the Seed
speaker is so low I cant hear what the ovos is saying.

**Comment by geetee24** [comment]
1. what is the wake work for Ovos?
2. ho do i set the volume for the seed2 speaker?

**Reply by builderjer** [reply]
WW is "hey mycroft"
Set the volume with "hey mycroft max volume"

**Reply by builderjer** [reply]
what version of the seeed speaker are you using?

**Comment by geetee24** [comment]
how do i know what version?

**Reply by builderjer** [reply]
what does it look like?
is it 2/4/6/8 mic?
is it USB or a HAT?

**Comment by geetee24** [comment]
2 mics.

alsa mixer does not list it.

**Comment by geetee24** [comment]
i type ovos-say-to......

and unit speaks that vol was set to max value BUT THE ACTUAL volume did not
change.   ALSAMIXER does not list my device

**Reply by builderjer** [reply]
That image may not have the update
run the command 
`pip install -U --pre ovos-core`
When that finishes, restart ovos with
`systemctl --user restart ovos`

You can see the tail of all the logs with the command `ologs`

**Reply by builderjer** [reply]
what does `pactl info` show

**Comment by geetee24** [comment]
hat 2 mic

**Reply by builderjer** [reply]
are you using the 3.5mm output on the board, or the speaker plug, or a USB speaker

**Comment by geetee24** [comment]
speaker plug on the hat

**Reply by builderjer** [reply]
That one works by default for me.  I have to set the 3.5mm out in alsamixer

in alsamixer select `f-6` and see if your device is there.  It will not be called "seeed", but `wm8960` or something?  Check the volume there

**Comment by geetee24** [comment]
I cant hear the prompt for the volume is so low.

**Comment by geetee24** [comment]
it does not do that.    i simply boots  and lists venv ovos shell commands
menu.

what the heck is going on ?

**Reply by builderjer** [reply]
That is what it should do.  This is a headless image not meant to be run with a screen.  Is the volume the only issue?

**Comment by geetee24** [comment]
alvsamixer hangs in the venv

if i deactivate the venv, alsamixer works fine.

how do i  reactivate the venv?

**Reply by builderjer** [reply]
`source ~/.venv/bin/activate`

**Reply by builderjer** [reply]
I am not sure what your issue is.  I can not reproduce it, and I am using the same hardware.

**Comment by geetee24** [comment]
volume is not loud.
ovos is not detecting the wake word
pressing F3 in alsamixer causes the mixer app to terminate.
how set us keyboard?

**Reply by builderjer** [reply]
you have something else going on.  US keyboard is automatic and everything else is working for me.

Are you running on a SD card? or USB or SSD?

maybe your image is corrupted, or your drive is bad.

I would suggest waiting 2 hours and downloading the latest image and trying that.

I am at a loss

**Comment by geetee24** [comment]
all is fine on my end.

**Comment by geetee24** [comment]
how do i file copy my custom MC skills into OVOS?  Where do I copy them too?

old MC had a UI that showed microphone level and real time parsing of
skills, speaking etc.  Ovos seems to have no such UI.  Is that correct?

**Reply by builderjer** [reply]
`ovos-cli-client` has been archived.  The mic level indicator does not work with dinkum-listener, and log locations have been changed to where they are not reliable for `ovos-cli-client`.

The suggested way is to use a couple of terminals, on with `ovos-simple-cli` and another "tailing" the logs...`tail -f ~/.local/state/mycroft/skills.log`

**Reply by builderjer** [reply]
Skills should be pip installable.

**Reply by geetee24** [reply]
the command ovos-simple-cli does not show me logs.  its shows SAYS:

please advise

**Comment by geetee24** [comment]
I can’t pip install it. For all my files are local. I want to copy them
from my old mc  folder to the new location on OVOS but I do not know where
I can drag them to and then how do I actually see them being compiled

**Reply by builderjer** [reply]
If you create a setup.py file, it can be installed locally without being on PyPi.

The old way, putting them in `~/.local/share/mycroft/skills` will NOT be supported in ovos-core 0.0.8

**Comment by geetee24** [comment]
huh?  what version of core did i just install?   I dont care about 0.0.8.
for now I want to copy them i

**Reply by JarbasAl** [reply]
ovos is currently in `0.0.8 alpha` and thats what the installer and our images ship

we do not provide support for older versions, long term support starts only with version `0.1.0`

there is no `stable` release yet!

**Comment by JarbasAl** [comment]
> The old way, putting them in `~/.local/share/mycroft/skills` will NOT be supported in ovos-core 0.0.8

actually it will until version `0.1.0` :)

**Comment by geetee24** [comment]
thank you.

can you post instructions on how to:
1. learn of new OVOS releases
2. How to install the OVOS code over the existing code (rasovos)

**Comment by geetee24** [comment]
My unit after reboot also complains about needing to be registered and it
actually says that roughly every five minutes forever and so I basically
determined that the backend solution that we’re talking about just doesn’t
work – sadly

**Reply by builderjer** [reply]
I created a public backend that may work. 

https://ovospublicbackend.ziggyai.online

More than welcome to give that address a shot

**Comment by geetee24** [comment]
suddenly my unit (stock rpi ovos image)   show

/dev/root is 100% full when it is just a rpi ovos standard image. How find
the error files? How recover?   What bug did i trigger?

**Reply by builderjer** [reply]
Did you change where the log files go? They can fill up a disk quickly

**Reply by geetee24** [reply]
i dragged my mycroft skill into the ovos folder but ovos doesnt appear to process the skill when i speak the intent of my skill.   where can i see the log of Ovos processing my new skill like I saw in the old mycroft UI

**Comment by geetee24** [comment]
![IMG002](https://github.com/OpenVoiceOS/ovos-buildroot/assets/6679119/c171e2a9-8fb5-4424-9193-82be46fb13a5)
no i did not.  where are they?
see attached image.  does this look right?


$ df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/root       5.1G  4.9G     0 100% /                 <----------------------- the issue
devtmpfs        1.7G     0  1.7G   0% /dev
tmpfs           1.9G     0  1.9G   0% /dev/shm
tmpfs           759M  1.1M  758M   1% /run
tmpfs           5.0M   12K  5.0M   1% /run/lock
tmpfs            20M     0   20M   0% /ramdisk
/dev/mmcblk0p1  510M   40M  471M   8% /boot/firmware
tmpfs           380M  4.0K  380M   1% /run/user/1000


suddenly my unit (stock rpi ovos image)   shows

/dev/root is 100% full when it is just a rpi ovos standard image. How find
the error files? How recover?   What bug did i trigger?

the image from OVOS for my rpi is a 4gb image.  my sdcard is 32gb.   how do
I resize

Help.............................................................

**Reply by builderjer** [reply]
Im not sure what image you are using, but the raspOVOS images resize the disk automatically.

**Comment by geetee24** [comment]
how can I get OVOS to speak the temp in fahrenheit?   editing
web_cache.json and mycroft.conf - both not work.

nope:
{
  "system_units": "imperial",
  "location": {
    "city": {
.....

and adding "  "system_units": "imperial","  to mycroft.conf also not work

**Reply by builderjer** [reply]
it is `"system_unit": "imperial"`

**Comment by geetee24** [comment]
I thought the latest image you gave me is 0.0.8

and you told me I could copy my files into that folder - which didnt work.
NOW what please?

**Comment by geetee24** [comment]
I’m using the image that you spoke of. And it absolutely did not resize the
disk automatically I had to manually do it. The image was stuck at 4 GB and
when I resized it it went up to 32 GB the size of my actual SD card.
